plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/util/expressionExt.kt

```diff
@@ -55,8 +54,9 @@ fun KtNamedDeclaration.nameIdentifierTextRangeInThis(): TextRange? = nameIdentif
 fun PsiElement.hasComments(): Boolean = anyDescendantOfType<PsiComment>()
 
 fun KtDotQualifiedExpression.hasNotReceiver(): Boolean {
-    val element = getQualifiedElementSelector()?.mainReference?.resolve() ?: return false
-    return element is KtClassOrObject ||
+    val selector = getQualifiedElementSelector() ?: return false
+    val element = selector.mainReference?.resolve() ?: return false
+    return element is KtClassOrObject && selector.text !in listOf("copy", "toString", "hashCode", "equals") ||
```

Comment:
I don't think what it should be fixed by text search – It can break valid cases. Probably `selector.resolveToCall()?.resultingDescriptor` can help here